enhancements/olm/service-account-deprecation.md

```diff
@@ -0,0 +1,262 @@
+
+- The operator-controller will **no longer impersonate the provided ServiceAccount**.
+- Instead, it will use **its own identity** (via its default `ServiceAccount` assigned to the controller) for all API calls and reconciliation.
+- `.spec.serviceAccount` will remain **optional but ignored**, and marked as deprecated with the intent to remove the field entirely in the future.
+
+
+### Goals
+
+- Make `.spec.serviceAccount` field **optional** in the `ClusterExtension` API.
+- Update the controller to **ignore** the field during reconciliation.
+- Log a warning when `.spec.serviceAccount` is set.
+- Provide a deprecation and removal plan.
+
+### Non-Goals
+
+- Replacing `serviceAccount` with another RBAC mechanism  
+- Managing permissions or pre-flight RBAC validation 
+
+## Proposal
+
+### API Changes
+
+Update the `ClusterExtension` API to mark `.spec.serviceAccount` as:
+
+- Optional
+- Deprecated via struct tags and documentation
+- Marked as deprecated in the CRD schema using OpenAPI extensions:
+  - x-kubernetes-deprecated: true
+  - Include appropriate description to warn users the field is ignored
+
+Also, update CRD validation schema accordingly. 
+- This will be done via OpenAPI `x-kubernetes-deprecated: true` annotation in the CRD.
+
+**Example:**
+
+```go
+// Before (current)
+type ClusterExtensionSpec struct {
+    ...
+    ServiceAccount ClusterExtensionServiceAccount `json:"serviceAccount"`
+    ...
+}
+```
+
+```go
+// After (deprecated and optional)
+type ClusterExtensionSpec struct {
+    ...
+    // Deprecated: This field is ignored and will be removed in a future release.
+    // +optional
+    // +kubebuilder:validation:XValidation:message="serviceAccount is deprecated and ignored"
+    ServiceAccount *ClusterExtensionServiceAccount `json:"serviceAccount,omitempty"`
+    ...
+}
+```
+
+**YAML Example:**
+
+```yaml
+# Before (serviceAccount was required)
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd-extension
+spec:
+  installNamespace: argocd
+  packageName: argocd-operator
+  version: 0.6.0
+  serviceAccount:
+    name: argocd-installer
+```
+
+```yaml
+# After (field is deprecated, optional, and therefore ignored and removed)
+apiVersion: olm.operatorframework.io/v1alpha1
+kind: ClusterExtension
+metadata:
+  name: argocd-extension
+spec:
+  installNamespace: argocd
+  packageName: argocd-operator
+  version: 0.6.0
+  # serviceAccount is now deprecated and has no effect
+```
+
+### Controller Logic Changes
+
+- Remove all logic that: 
+  - Token Acquisition: Eliminate use of TokenRequest API to fetch short-lived tokens for user-provided ServiceAccounts.
+  - Rest Config Mapping: Remove the ServiceAccountRestConfigMapper, which dynamically generated rest.Config objects for impersonation.
+  - Synthetic Permissions: Remove conditional logic for SyntheticPermissions that depended on impersonated clients.
```

Comment:
How do "Synthetic Permissions" play a role in service account based permissions? What are "Synthetic Permissions"?